src/plonk/lookup.rs

```diff
@@ -33,6 +34,15 @@ impl Argument {
         //
         // degree 2:
         // (a′(X)−s′(X))⋅(a′(X)−a′(\omega{-1} X)) = 0
-        3
+        let mut input_degree = 1;
+        for expr in self.input_expressions.iter() {
+            input_degree = std::cmp::max(input_degree, expr.degree());
+        }
+        let mut table_degree = 1;
+        for expr in self.table_expressions.iter() {
+            table_degree = std::cmp::max(table_degree, expr.degree());
+        }
+
+        1 + input_degree + table_degree
```

Comment:
Technically we could have expressions which are constants, and so `input_degree` and `table_degree` should start out as zero. If they did, this would need to change to `std::cmp::max(1 + input_degree + table_degree, 2)` where the 2 comes from the other gate expressions.

However, that would be weird and for now we can leave it like this.